interpreter/shared/src/main/scala/sigmastate/eval/Extensions.scala

```diff
@@ -120,6 +120,24 @@ object Extensions {
       }
     }
 
+    def insertOrUpdate(
+                entries: Coll[(Coll[Byte], Coll[Byte])],
+                proof: Coll[Byte]): Option[AvlTree] = {
+      if (!tree.isInsertAllowed || !tree.isUpdateAllowed) {
```

Comment:
Both flags are required, however for each given tree it will either be insert or update (but not both).
Is it possible to make it more flexible and require the flag only if the corresponding operation is happening for a given tree.